### PR TITLE
Add riverpod tool

### PR DESCRIPTION
**Pull request by adsonpleal**
This PR adds support to `Riverpod`, the code is heavily based on the current `Provider` implementation. It uses provider's `InstanceViewer` widget to show the state values.

https://user-images.githubusercontent.com/11666470/175167035-4027eb48-7f7b-44e6-bb83-af7e8d8feaaf.mp4

The discussion about this solution can be [found at riverpod's repo](https://github.com/rrousselGit/riverpod/issues/1039#issuecomment-1163572176).

***TODO: this PR is still a Draft, the tests will be added before opening it for real.***

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/devtools_app/test/fixtures/riverpod_app/pubspec.yaml [hunk 1]
```diff
@@ -9,7 +9,8 @@ environment:
 dependencies:
   flutter:
     sdk: flutter
-  flutter_riverpod: ^2.0.0-dev.5
+  flutter_riverpod: 
+    path: /Users/adson.leal/dev/personal/riverpod/packages/flutter_riverpod # TODO: need to use an actual version
```
Comment (adsonpleal) [inline packages/devtools_app/test/fixtures/riverpod_app/pubspec.yaml:13]: @rrousselGit we need to release a version first, before merging this PR.

**Comment by adsonpleal** [comment]
@rrousselGit Just added tests, this PR is ready to be reviewed, but as we need the [riverpod](https://github.com/rrousselGit/riverpod/pull/1471) one to be merged first I'll leave this one as `draft` for now.

File: packages/devtools_app/test/fixtures/riverpod_app/pubspec.yaml (hunk 1 above)
Comment (rrousselGit) [reply packages/devtools_app/test/fixtures/riverpod_app/pubspec.yaml:13]: That's not a problem. It'll happen later

**Comment by rrousselGit** [comment]
That's great, thanks for your work!

I'll try playing around with it this w-e to see.

**Comment by rrousselGit** [comment]
This is looking great to me.

Any thoughts @jacob314?

**Comment by rrousselGit** [comment]
If that's good for you, I can merge the associated PR adding dart:developer events to Riverpod, then we can remove the git dependency.

**Comment by rrousselGit** [comment]
While we're waiting for a review, it'd be great to add support for the upcoming code-generator

Fancy supporting it?

**Comment by adsonpleal** [comment]
You mean `Riverpod 2.0`?

**Comment by rrousselGit** [comment]
I mean the "Riverpod_generator" project in the master branch of Riverpod

**Comment by adsonpleal** [comment]
I'll take a look, if I have questions  I'll talk to you.

**Comment by cedvdb** [comment]
How does opt-in work here ? I'm confused why this PR is made on the devtool itself as riverpod has nothing to do with the dev tools.

I'd have assumed a riverpod_devtools or some external package. Same if I want to create my own dev tool extension.

**Comment by kenzieschmoll** [comment]
Since DevTools does not yet support plugins or extensions, we're going to close this for now. DevTools plugin support is something we are hoping to work on later this year ([here](https://docs.google.com/document/d/1BWX8YQ962Vsx-EUDuDHRG7RX94fJlZSDRu299YRwasE/edit) is an existing design proposal for this). Converting the existing package:provider screen as well as this riverpod screen to a plugin would be great first use cases to work on when plugin support is added. We will definitely reach out when this support is available. Thank you for the contribution!

**Comment by kenzieschmoll** [comment]
Hello @adsonpleal and @rrousselGit - now that support for [DevTools extensions](https://pub.dev/packages/devtools_extensions) has officially landed, do you have interest in re-landing this PR on the [riverpod repo](https://github.com/rrousselGit/riverpod) directly as a DevTools extension? As an example, the Provider related code from DevTools was migrated to a DevTools extension in `package:provider` [here](https://github.com/rrousselGit/provider/pull/832), and published as part of [6.1.0-dev.1](https://pub.dev/packages/provider/versions/6.1.0-dev.1)

@rrousselGit can you provide some guidance on which [riverpod package](https://github.com/rrousselGit/riverpod/tree/master/packages) the extension should live in (I would guess `riverpod`?)

**Comment by rrousselGit** [comment]
I'm already working on it :)

**Pull request closed without merging**